apps/crn-server/src/controllers/research-outputs.ts

```diff
@@ -30,7 +31,10 @@ export default class ResearchOutputs implements ResearchOutputController {
     private externalAuthorDataProvider: ExternalAuthorDataProvider,
   ) {}
 
-  async fetchById(researchOutputId: string): Promise<ResearchOutputResponse> {
+  async fetchById(
+    researchOutputId: string,
+    loggedInUser?: UserResponse,
```

Comment:
although doing this inside the controller allows you to skip some of the parsing logic i think the pattern is that we check the permissions in the routes rather than controllers, im not sure if that's valid but this is how we have done it until now (though I think if we were to reuse same controller methods in different routes same rules would apply so maybe we should move it to controllers at some point)